examples/js/geometries/TeapotGeometry.js

```diff
@@ -0,0 +1,634 @@
+			}
+
+			// step along, get points, and output
+			for ( sstep = 0 ; sstep <= this.segments ; sstep++ ) {
+				s = sstep / this.segments;
+
+				for ( tstep = 0 ; tstep <= this.segments ; tstep++ ) {
+					t = tstep / this.segments;
+
+					// point from basis
+					// get power vectors and their derivatives
+					for ( p = 4, sval = tval = 1.0 ; p-- ; ) {
+						sp[p] = sval ;
+						tp[p] = tval ;
+						sval *= s ;
+						tval *= t ;
+
+						if ( p === 3 ) {
+							dsp[p] = dtp[p] = 0.0 ;
+							dsval = dtval = 1.0 ;
+						} else {
+							dsp[p] = dsval * (3-p) ;
+							dtp[p] = dtval * (3-p) ;
+							dsval *= s ;
+							dtval *= t ;
+						}
+					}
+
+					vsp = new THREE.Vector4( sp[0], sp[1], sp[2], sp[3] );
+					vtp = new THREE.Vector4( tp[0], tp[1], tp[2], tp[3] );
+					vdsp = new THREE.Vector4( dsp[0], dsp[1], dsp[2], dsp[3] );
+					vdtp = new THREE.Vector4( dtp[0], dtp[1], dtp[2], dtp[3] );
+
+					// do for x,y,z
+					for ( i = 0 ; i < 3 ; i++ ) {
+						// multiply power vectors times matrix to get value
+						tcoord = vsp.clone();
+						tcoord.applyMatrix4( mgm[i] );
+						vert[i] = tcoord.dot( vtp );
+
+						// get s and t tangent vectors
+						tcoord = vdsp.clone();
+						tcoord.applyMatrix4( mgm[i] );
+						sdir[i] = tcoord.dot( vtp ) ;
+
+						tcoord = vsp.clone();
+						tcoord.applyMatrix4( mgm[i] );
+						tdir[i] = tcoord.dot( vdtp ) ;
+					}
+
+					// find normal
+					vsdir = new THREE.Vector3( sdir[0], sdir[1], sdir[2] );
+					vtdir = new THREE.Vector3( tdir[0], tdir[1], tdir[2] );
+					norm.crossVectors( vtdir, vsdir );
+					norm.normalize();
+
+					// rotate on X axis
+					normOut = new THREE.Vector3( norm.x, norm.z, -norm.y );
+
+					// if X and Z length is 0, at the cusp, so point the normal up or down, depending on patch number
+					if ( vert[0] === 0 && vert[1] === 0 )
+					{
+						// if above the middle of the teapot, normal points up, else down
+						normOut.set( 0, vert[2] > maxHeight2 ? 1 : -1, 0 );
+					}
+					normals.push( normOut );
+
+					uvs.push( new THREE.Vector2( 1-t, 1-s ) );
+
+					// three.js uses Y up, the code makes Z up, so time for a trick:
+					// rotate on X axis, and offset down on Y axis so object ranges from -1 to 1 in Y
+					vertOut = new THREE.Vector3( trueSize*vert[0], trueSize*(vert[2] - maxHeight2), -trueSize*vert[1] );
+
+					this.vertices.push( vertOut );
+
+				}
+			}
+
+			// save the faces
+			for ( sstep = 0 ; sstep < this.segments ; sstep++ ) {
+				for ( tstep = 0 ; tstep < this.segments ; tstep++ ) {
+					v1 = surfCount * vertPerRow * vertPerRow + sstep * vertPerRow + tstep;
+					v2 = v1 + 1;
+					v3 = v2 + vertPerRow;
+					v4 = v1 + vertPerRow;
+
+					if ( notDegenerate ( this.vertices[v1], this.vertices[v2], this.vertices[v3] ) ) {
+						this.faces.push( new THREE.Face3( v1, v2, v3, [ normals[v1], normals[v2], normals[v3] ] ) );
+						this.faceVertexUvs[ 0 ].push( [ uvs[v1], uvs[v2], uvs[v3] ] );
+					}
```

Comment:
```javascript
this.faces.push( new THREE.Face3( v1, v2, v3, [ normals[ v1 ].clone(), normals[ v2 ].clone(), normals[ v3 ].clone() ] ) );
this.faceVertexUvs[ 0 ].push( [ uvs[ v1 ].clone(), uvs[ v2 ].clone(), uvs[ v3 ].clone() ] );
```
Normals and UVs cannot be shared. You can see the consequences of sharing if you call `geometry.applyMatrix( matrix )`.

You can use [mrdoobapproves](http://zz85.github.io/mrdoobapproves/) if you want help with your formatting.